.github/workflows/pythonpackage.yml

```diff
@@ -12,8 +12,14 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 1
+    
+    - name: Fetch main branch
+      run: |
+        git fetch origin main:refs/remotes/origin/main
```

Comment:
Will this workflow always be on main only?

Reply:
Yes we will check the modification of _deprecated rules on `main` only

Reply:
I think it's a bit confusing because the pipeline will trigger on other branches as well -- https://github.com/elastic/detection-rules/blob/bc4b289cab8099bcf5d1e36929ebd6aff9f76e01/.github/workflows/pythonpackage.yml#L5

should that be adjusted?

Or, alternatively, we can specify a condition for a specific job with
```
if: github.ref == 'refs/heads/main'
```
condition

Reply:
Discussed on call and cleared the implementation!